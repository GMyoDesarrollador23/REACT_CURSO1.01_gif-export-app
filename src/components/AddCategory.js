import React, { useState } from "react";
import PropTypes from "prop-types";

export const AddCategory = ({ setCategories }) => {
   const [inputValue, setInputValue] = useState("");

   const handleInputValue = (e) => {
      setInputValue(e.target.value);
   };

   const handleSubmit = (e) => {
      e.preventDefault();
      // console.log("handleSubmit", inputValue);

      if (inputValue.trim().length > 2) {
         setCategories((cats) => [inputValue, ...cats]);

         setInputValue("");
      }
   };

   return (
      <form onSubmit={handleSubmit}>
         <p>{inputValue}</p>
         <input
            type="text"
            value={inputValue}
            onChange={handleInputValue}
         />
      </form>
   );
};

AddCategory.protoTypes = {
   setCategories: PropTypes.func.isRequired,
};
