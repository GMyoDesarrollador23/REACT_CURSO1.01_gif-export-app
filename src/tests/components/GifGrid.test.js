import React from "react";
import { shallow } from "enzyme";
import "@testing-library/jest-dom";

import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";

jest.mock("../../hooks/useFetchGifs");

describe("preubas sobre el <GifGrid/>", () => {
   const category = "";

   // -----------------------------------------------------
   test("debe de mostrarso correctamente", () => {
      useFetchGifs.mockReturnValue({
         data: [],
         loading: true,
      });

      const wrapper = shallow(
         <GifGrid category={category} />
      );
      expect(wrapper).toMatchSnapshot();
   });

   // -----------------------------------------------------
   test("debe de mostrar items cuando se cargan las imajenes useFetchGifs", () => {
      const gits = [
         {
            id: "ABC",
            url: "https://cualquiercasa.jpg",
            title: "algun titulo",
         },
         {
            id: "123",
            url: "https://cualquiercasa.jpg",
            title: "algun titulo",
         },
      ];

      useFetchGifs.mockReturnValue({
         data: gits,
         loading: false,
      });

      const wrapper = shallow(
         <GifGrid category={category} />
      );

      expect(wrapper).toMatchSnapshot();
      expect(wrapper.find("p").exists()).toBe(false);
      expect(wrapper.find("GifGridItem").length).toBe(
         gits.length
      );
   });
});
