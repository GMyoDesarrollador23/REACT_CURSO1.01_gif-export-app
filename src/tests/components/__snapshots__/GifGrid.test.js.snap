// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`preubas sobre el <GifGrid/> debe de mostrar items cuando se cargan las imajenes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  />
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="algun titulo"
      url="https://cualquiercasa.jpg"
    />
    <GifGridItem
      id="123"
      key="123"
      title="algun titulo"
      url="https://cualquiercasa.jpg"
    />
  </div>
</Fragment>
`;

exports[`preubas sobre el <GifGrid/> debe de mostrarso correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  />
  <p
    className="animate__animated animate__heartBeat"
  >
    Cargando...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
